src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx

```diff
@@ -1,24 +1,115 @@
+import { v4 as uuidv4 } from 'uuid';
 
 import { Svg } from 'shared/components';
 import { StyledTitleMedium, theme } from 'shared/styles';
 import { BuilderContainer } from 'shared/features';
 import { useBreadcrumbs } from 'shared/hooks';
-import { Item } from 'modules/Builder/components';
-import { ActivityFlow as ActivityFlowType } from 'shared/state';
+import { Item, ItemUiType } from 'modules/Builder/components';
+import { page } from 'resources';
+import { getNewActivityFlow } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.utils';
+import {
+  ActivityFlowFormValues,
+  AppletFormValues,
+} from 'modules/Builder/pages/BuilderApplet/BuilderApplet.types';
 
-import { getActions } from './ActivityFlow.const';
+import { DeleteFlowModal } from './DeleteFlowModal';
+import { getFlowsItemActions } from './ActivityFlow.utils';
 import { StyledAdd, StyledAddWrapper } from './ActivityFlow.styles';
 import { ActivityFlowHeader } from './ActivityFlowHeader';
 
 export const ActivityFlow = () => {
+  const [, setDuplicateIndexes] = useState<Record<string, number> | null>(null);
+  const [flowToDeleteData, setFlowToDeleteData] = useState<{
+    id: string;
+    index: number;
+    name: string;
+  } | null>(null);
   const { t } = useTranslation('app');
-  const { watch } = useFormContext();
+  const { watch, control, getFieldState } = useFormContext<AppletFormValues>();
+  const { appletId } = useParams();
+  const navigate = useNavigate();
 
-  const activityFlows: (Omit<ActivityFlowType, 'description'> & { description: string })[] =
-    watch('activityFlows');
+  const activityFlows: AppletFormValues['activityFlows'] = watch('activityFlows');
+
+  const {
+    remove: removeActivityFlow,
+    append: appendActivityFlow,
+    insert: insertActivityFlow,
+    update: updateActivityFlow,
+  } = useFieldArray({
+    control,
+    name: 'activityFlows',
+  });
+
+  const errors = activityFlows?.reduce(
+    (err: Record<string, boolean>, _: ActivityFlowFormValues, index: number) => ({
+      ...err,
+      [`activityFlows.${index}`]: !!getFieldState(`activityFlows.${index}`).error,
+    }),
+    {},
+  );
+
+  const handleFlowDelete = () => {
+    if (!flowToDeleteData) return;
+
+    removeActivityFlow(flowToDeleteData.index);
+    setFlowToDeleteData(null);
+  };
+
+  const handleEditActivityFlow = (activityFlowId: string) =>
+    navigate(
+      generatePath(page.builderAppletActivityFlowItem, {
+        appletId,
+        activityFlowId,
+      }),
+    );
+
+  const handleAddActivityFlow = (positionToAdd?: number) => {
+    const newActivityFlow = getNewActivityFlow();
+
+    if (positionToAdd) {
+      insertActivityFlow(positionToAdd, newActivityFlow);
+    } else {
+      appendActivityFlow(newActivityFlow);
+    }
+    handleEditActivityFlow(newActivityFlow.id);
+  };
+
+  const handleDuplicateActivityFlow = (index: number) => {
+    const duplicatedItem = activityFlows[index];
+    setDuplicateIndexes((prevState) => {
```

Comment:
do we really need `duplicateIndexes` and `setDuplicateIndexes`?

Reply:
discussed with BA, indexes are needed